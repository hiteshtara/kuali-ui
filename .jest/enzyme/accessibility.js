/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import assert from 'assert'
import { findDOMNode } from 'react-dom'
import { mount } from 'enzyme'
import axeCore from 'axe-core'

export async function assertComponentIsAccessible (component, axeConfig = {}) {
  const container = document.createElement('div')
  document.body.appendChild(container)

  const wrapper = mount(component, { attachTo: container })
  const nodeUnderTest = findDOMNode(wrapper.instance())

  const { violations = [] } = await axeCore.run(nodeUnderTest, axeConfig)

  for (const violation of violations) {
    const { help } = violation

    for (const node of violation.nodes) {
      const { target } = node
      assert.ok(false, `${help} on element ${target}`)
    }
  }

  document.body.removeChild(container)
}
