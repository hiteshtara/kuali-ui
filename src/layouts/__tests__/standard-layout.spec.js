/* Copyright Â© 2016 Kuali, Inc. - All Rights Reserved
 * You may use and modify this code under the terms of the Kuali, Inc.
 * Pre-Release License Agreement. You may not distribute it.
 *
 * You should have received a copy of the Kuali, Inc. Pre-Release License
 * Agreement with this file. If not, please write to license@kuali.co.
 */

import React from 'react'
import { shallow } from 'enzyme'

import { Header, NavList, NavListItem } from '../../'
import { Sidebar, StandardLayout, TitleBar } from '..'

const noop = () => {}

describe('StandardLayout Component', () => {
  const header = (
    <Header
      menuItems={[]}
      module={{ name: 'App Name' }}
      onBrandSelected={noop}
      user={{ name: 'User Name' }}
    />
  )

  const navList = (
    <NavList>
      <NavListItem iconName='create'>
        Pellentesque habitant morbi tristique senectus
      </NavListItem>
    </NavList>
  )

  const sidebar = (
    <Sidebar>
      <p>sidebar component</p>
    </Sidebar>
  )

  const titleBar = <TitleBar>Page Title</TitleBar>

  it('should render', () => {
    const wrapper = shallow(
      <StandardLayout
        header={header}
        navList={navList}
        sidebar={sidebar}
        titleBar={titleBar}
      >
        <p>body content</p>
      </StandardLayout>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render without a nav list', () => {
    const wrapper = shallow(
      <StandardLayout header={header} sidebar={sidebar} titleBar={titleBar}>
        <p>body content</p>
      </StandardLayout>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
