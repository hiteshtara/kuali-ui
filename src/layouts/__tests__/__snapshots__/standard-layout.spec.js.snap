// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardLayout Component should render 1`] = `
<div
  className="layout layout--standard"
>
  <div
    id="skipmenu"
  >
    <a
      className="layout__skippy"
      href="#nav"
    >
      Skip directly to navigation
    </a>
    <a
      className="layout__skippy"
      href="#mainContent"
    >
      Skip directly to main content
    </a>
  </div>
  <Header
    Logo={<BrandLogo />}
    menuItems={Array []}
    module={
      Object {
        "name": "App Name",
      }
    }
    onBrandSelected={[Function]}
    user={
      Object {
        "name": "User Name",
      }
    }
  />
  <div
    className="layout__body"
  >
    <nav
      className="layout__nav"
      id="nav"
    >
      <NavList
        collapsed={false}
        nested={false}
      >
        <withInk(NavListItem)
          active={false}
          className="nav-list__item--hide"
          container={[Function]}
          defaultNestedCollapsed={false}
          iconName="clear"
          onClick={[Function]}
        >
          Hide Menu
        </withInk(NavListItem)>
      </NavList>
      <NavList
        collapsed={false}
        nested={false}
      >
        <withInk(NavListItem)
          active={false}
          container={[Function]}
          defaultNestedCollapsed={false}
          iconName="create"
        >
          Pellentesque habitant morbi tristique senectus
        </withInk(NavListItem)>
      </NavList>
      <FlatButton
        className="layout__nav-close"
        onClick={[Function]}
        variant="clear"
      >
        <Icon
          aria-hidden={true}
          iconClassName="material-icons"
          name="close"
          variant="default"
        />
      </FlatButton>
    </nav>
    <main
      className="layout__main"
    >
      <TitleBar
        hasNav={true}
        onToggleNavOverlay={[Function]}
      >
        Page Title
      </TitleBar>
      <div
        className="layout__content-wrapper"
      >
        <div
          className="layout__content"
          id="mainContent"
        >
          <p>
            body content
          </p>
        </div>
        <Sidebar
          className="layout__sidebar"
        >
          <p>
            sidebar component
          </p>
        </Sidebar>
      </div>
    </main>
  </div>
</div>
`;

exports[`StandardLayout Component should render without a nav list 1`] = `
<div
  className="layout layout--standard"
>
  <div
    id="skipmenu"
  >
    <a
      className="layout__skippy"
      href="#nav"
    >
      Skip directly to navigation
    </a>
    <a
      className="layout__skippy"
      href="#mainContent"
    >
      Skip directly to main content
    </a>
  </div>
  <Header
    Logo={<BrandLogo />}
    menuItems={Array []}
    module={
      Object {
        "name": "App Name",
      }
    }
    onBrandSelected={[Function]}
    user={
      Object {
        "name": "User Name",
      }
    }
  />
  <div
    className="layout__body"
  >
    <main
      className="layout__main"
    >
      <TitleBar
        hasNav={false}
        onToggleNavOverlay={[Function]}
      >
        Page Title
      </TitleBar>
      <div
        className="layout__content-wrapper"
      >
        <div
          className="layout__content"
          id="mainContent"
        >
          <p>
            body content
          </p>
        </div>
        <Sidebar
          className="layout__sidebar"
        >
          <p>
            sidebar component
          </p>
        </Sidebar>
      </div>
    </main>
  </div>
</div>
`;
